gtsam/navigation/Gal3ImuEKF.cpp

```diff
@@ -0,0 +1,80 @@
+/* ----------------------------------------------------------------------------
+
+ * GTSAM Copyright 2010, Georgia Tech Research Corporation,
+ * Atlanta, Georgia 30332-0415
+ * All Rights Reserved
+ * Authors: Frank Dellaert, et al. (see THANKS for the full author list)
+
+ * See LICENSE for the license information
+
+ * -------------------------------------------------------------------------- */
+
+/**
+ * @file  Gal3ImuEKF.cpp
+ * @brief Extended Kalman Filter derived class for IMU-driven Gal3.
+ *
+ * @date  September 2025
+ * @authors Scott Baker
+ */
+
+#include <gtsam/base/Matrix.h>
+#include <gtsam/base/numericalDerivative.h>
+#include <gtsam/navigation/Gal3ImuEKF.h>
+
+namespace gtsam {
+
+Gal3ImuEKF::Gal3ImuEKF(const Gal3& X0, const Covariance& P0,
+                               const std::shared_ptr<PreintegrationParams>& p)
+    : Base(X0, P0), params_(p) {
+  // Build process noise Q_ = block_diag(Cg, Ci, Ca)
+  Q_.setZero();
+  Q_.template block<3, 3>(0, 0) = p->gyroscopeCovariance;
+  Q_.template block<3, 3>(3, 3) = p->integrationCovariance;
+  Q_.template block<3, 3>(6, 6) = p->accelerometerCovariance;
+}
+
+Gal3 Gal3ImuEKF::Dynamics(const Vector3& n_gravity, const Gal3& X,
+                                  const Vector3& omega_b, const Vector3& f_b,
+                                  double dt, OptionalJacobian<10, 10> A) {
+  if (dt <= 0.0) {
+    throw std::invalid_argument(
+        "Gal3ImuEKF::Dynamics: dt must be positive");
+  }
+
+  // Calculate W, phi, and U
+  const Gal3 W = Gravity(n_gravity, dt);
+  Gal3::AutonomousFlow phi{dt};  // Φ: velocity acts on position
```

Comment:
This should be a no-op automorphism, for now I'd just define that locally in this file. It is not needed in Gal3 and when we switch to InvariantEKF it'll be not needed at all.